### PR TITLE
Initial model setup

### PR DESCRIPTION
**Pull request by codemutatio**
**What does this PR do?**

- Application Model setup

- Database CRUD operation

### CODE REVIEW

File: server/controllers/aidControllers.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	}
+
+    // Create a aid
+    const aid = new Aid({
+        aidTitle: req.body.aidTitle, 
+        aidIntro: req.body.aidIntro,
+        aidDescription: req.body.aidDescription,
+        imageLink: req.body.imageLink,
+        videoLink: req.body.videoLink
+    });
```
Comment (EngrEric) [inline server/controllers/aidControllers.js]: destructure this code
`const { 
        aidTitle, 
        aidIntro,
        aidDescription,
       imageLink,
       videoLink } = req.body; `

**Review by EngrEric — COMMENTED** [review]
Looks good but aren't you leveraging the async/await paradigm.

**Comment by enyeInc** [comment]
@codemutatio Go through each one of your controllers and like eric said leverage `async/await` pattern (https://javascript.info/async-await).

File: server/controllers/aidControllers.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+const db = require("./promise").AidDb;
+import validateAidQueryText from '../validation/aid'
+
+
+const Aids = {
+    async create(req, res) {
+        console.log(req.body);
```
Comment (enyeInc) [inline server/controllers/aidControllers.js]: remove `console.log`

File: server/controllers/aidControllers.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            data: createdAid,
+          });
+        } catch (error) {
+          return res.status(400).send(error);
+        }
+    },
+	async getAll(req, res) {
+        const queryText = {};
+        try {
+            const foundAids = await db.find(queryText);
```
Comment (enyeInc) [inline server/controllers/aidControllers.js]: why are you passing an empty object to `db.find`?

File: server/controllers/aidControllers.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                message: 'Aidss retrieved successfully',
+                data: foundAids,
+            });
+        } catch (error) {
+            return res.status(400).send(error);
+        }
+	},
+	async getOne(req, res) {
+		const queryText = {
+			_id: req.params.aidId,
```
Comment (enyeInc) [inline server/controllers/aidControllers.js]: change this to `_id: req.params.id`. Change it in other parts of this file and other files

File: server/controllers/promise.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
+import Aid from  "../models/aid"
+import Report from "../models/report"
+
+class AidDb {
+    static find(param) {
+      return new Promise((resolve, reject) => {
```
Comment (enyeInc) [inline server/controllers/promise.js]: you do not need to wrap these functions in a promise, they are already promises.
```
return Aid.find(param);
```
is all you need to do. Go through this file and fix this

File: server/controllers/aidControllers.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+		}
+        const queryText = req.body;
+        try {
+          const createdAid = await db.create(queryText);
+          return res.status(201).send({
+            message: 'Aid successfully created',
+            data: createdAid,
+          });
+        } catch (error) {
+          return res.status(400).send(error);
```
Comment (enyeInc) [inline server/controllers/aidControllers.js]: the status code for errors in a server is `500` not `400` make this change in all your filles

**Comment by enyeInc** [comment]
@codemutatio this is almost ready to be merge. Make the final changes and additional use this module to define your error status code
- https://www.npmjs.com/package/http-status-codes

For example:
```
import HttpStatus from 'http-status-codes';
const { INTERNAL_SERVER_ERROR }  = HttpStatus;
...
catch(error) {
    return res.status(INTERNAL_SERVER_ERROR).send(error);
}
```

**Review by enyeInc — CHANGES_REQUESTED** [review]
Need to fix more syntax

File: server/controllers/reportControllers.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            return res.status(OK).send({
+              message: 'Report successfully deleted',
+              data: deletedReport,
+            });
+          } catch (error) {
+            return res.status(INTERNAL_SERVER_ERROR).send(error);
+          }
+    },
+
+
```
Comment (enyeInc) [inline server/controllers/reportControllers.js]: delete unused `lines 60-61`

**Comment by enyeInc** [comment]
@codemutatio Your code looks great just one last thing for you to add for it to be excellent. You need to standardize the data that is being returned from your backend. This is listed in the enye conventions. The standard that we use is JSend (https://github.com/omniti-labs/jsend). I left an example in the comments above. Adjust your endpoints to conform to this specification and you will have my approval to merge this down

File: server/controllers/aidControllers.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+		if (!isValid) {
+			return res.status(BAD_REQUEST).json(errors);
+		}
+        const queryText = req.body;
+        try {
+          const createdAid = await db.create(queryText);
+          return res.status(CREATED).send({
+            message: 'Aid successfully created',
+            data: createdAid,
+          });
```
Comment (enyeInc) [inline server/controllers/aidControllers.js]: Use JSend specification (https://github.com/omniti-labs/jsend) when sending out of your endpoints
```
...
const aid = await db.create(queryText);
return res.status(CREATED).send({
    status: 'success',
    data: { aid, message: 'Aid successfully created' },
});
...

**Review by iammiracle — COMMENTED** [review]
I think once the JSend is fixed, we can merge this.

File: server/controllers/aidControllers.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        try {
+            const updatedAid = await db.findOneAndUpdate(queryText, updateData);
+            if (!updatedAid) return res.status(NOT_FOUND).send({
+                status : 'error',
+                message : getStatusText(NOT_FOUND)
+            });
+            return res.status(OK).send({
+                status: 'success',
+                data: { UpdatedAid, message: 'Aid successfully updated' }
+            });
```
Comment (Way2nnadi) [inline server/controllers/aidControllers.js:92]: i gave you an example of how you should name your data variables and the overall syntax. Do not send an `UpdaredAid` or a `foundAid`, just send `aid` or `aids`. Also use single quotations not double. that should have been enforced in your eslint

File: server/controllers/aidControllers.js [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+        };
+        try{
+            const aid = await db.findOne(queryText);
+            if (!aid) return res.status(NOT_FOUND).send({
+                status : 'error',
+                message : getStatusText(NOT_FOUND)
+            });
+            return res.status(OK).send({
+                status: 'success',
+                data: {"Aid": aid}
```
Comment (Way2nnadi) [inline server/controllers/aidControllers.js]: `data: { aid }`

File: server/controllers/aidControllers.js [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+          });
+        }
+    },
+
+	async getAll(req, res) {
+        try {
+            const aids = await db.find();
+            return res.status(OK).send({
+                status: 'success',
+                data: {"Aids": aids}
```
Comment (Way2nnadi) [inline server/controllers/aidControllers.js]: In javascript first letters of keys in objects are lower case => `data: { aids }`

File: server/controllers/aidControllers.js [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+
+const Aids = {
+    async create(req, res) {
+        const { errors, isValid } = validateAidQueryText(req.body);
+
+		// Check Validation
+		if (!isValid) {
+			return res.status(BAD_REQUEST).send({
+                status : "fail",
+                data : { "title" : errors }
```
Comment (Way2nnadi) [inline server/controllers/aidControllers.js]: single quotations

File: server/controllers/aidControllers.js [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+            _id: req.params.id
+        };
+
+        const { errors, isValid } = validateAidQueryText(req.body);
+
+		// Check Validation
+		if (!isValid) {
+			return res.status(BAD_REQUEST).send({
+                status : "fail",
+                data : { "title" : errors }
```
Comment (Way2nnadi) [inline server/controllers/aidControllers.js]: title can not be errors just send the errors object => `data: { errors }`

File: server/controllers/reportControllers.js [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+      });
+    }
+  },
+
+	async getAll(req, res) {
+    try {
+      const reports = await db.find();
+      return res.status(OK).send({
+        status: 'success',
+        data: {"Reports": reports},
```
Comment (Way2nnadi) [inline server/controllers/reportControllers.js]: fix this

File: server/controllers/reportControllers.js [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+    };
+    try{
+      const report = await db.findOne(queryText);
+      if (!report) return res.status(NOT_FOUND).send({
+        status : 'error',
+        message : getStatusText(NOT_FOUND)
+      });
+      return res.status(OK).send({
+        status: 'success',
+        data: {"Report": report}
```
Comment (Way2nnadi) [inline server/controllers/reportControllers.js]: fix this

File: server/controllers/aidControllers.js [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+            });
+            return res.status(OK).send({
+                status: 'success',
+                data: {"Aid": aid}
+            });
+          }
+        catch(err){
+            return res.status(INTERNAL_SERVER_ERROR).send({
+                status : 'error',
+                message : error
```
Comment (Way2nnadi) [inline server/controllers/aidControllers.js]: where are you getting this `error` variable from? check this in this file and the other files, you have an syntax error

File: server/controllers/aidControllers.js [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+                message : getStatusText(NOT_FOUND)
+            });
+            return res.status(OK).send({
+                status: 'success',
+                data: { UpdatedAid, message: 'Aid successfully updated' }
+            });
+        } catch (error) {
+            return res.status(INTERNAL_SERVER_ERROR).send({
+                status : 'error',
+                message : error
```
Comment (Way2nnadi) [inline server/controllers/aidControllers.js]: `error` is an object so i would say do something like this
```
...
    status : 'error',
    message : getStatusText(INTERNAL_SERVER_ERROR),
    error,
...
```
Fix this in this file and the others

**Review by enyeInc — APPROVED** [review]
Looks good to me

**Review by iammiracle — APPROVED** [review]

**Pull request merged**